atproto/identity/handle.go

```diff
@@ -138,22 +138,29 @@ func (d *BaseDirectory) ResolveHandleWellKnown(ctx context.Context, handle synta
 		var dnsErr *net.DNSError
 		if errors.As(err, &dnsErr) {
 			if dnsErr.IsNotFound {
-				return "", ErrHandleNotFound
+				return "", fmt.Errorf("%w: DNS NXDOMAIN for %s", ErrHandleNotFound, handle)
 			}
 		}
-		return "", fmt.Errorf("failed to resolve handle (%s) through HTTP well-known route: %s", handle, err)
+		slog.Info("HTTP request failed for well-known handle resolution", "handle", handle, "err", err)
+		return "", fmt.Errorf("%w: HTTP request error: %w", ErrHandleResolutionFailed, err)
+	}
+	if resp.StatusCode == http.StatusNotFound {
+		return "", fmt.Errorf("%w: HTTP 404 for %s", ErrHandleNotFound, handle)
 	}
 	if resp.StatusCode != http.StatusOK {
-		return "", fmt.Errorf("failed to resolve handle (%s) through HTTP well-known route: status=%d", handle, resp.StatusCode)
+		slog.Info("non-success HTTP response for well-known handle resolution", "handle", handle, "status_code", resp.StatusCode)
```

Comment:
We might want this to be debug level maybe? I could see this triggering a ton since we race resolutions.